src/main/groovy/com/pstehlik/groovy/gelf4j/appender/Gelf4JAppender.groovy

```diff
@@ -76,83 +72,33 @@ extends AppenderSkeleton {
   }
 
   Map createGelfMapFromLoggingEvent(LoggingEvent loggingEvent) {
-    StringBuilder fullMessage = new StringBuilder(100 * this.getMaxLoggedLines())
-    String[] throwableAsString
-
-    //detecting if the actual log-message is a throwable - because if so, might want the whole stacktrace
-    if(loggingEvent.message instanceof Throwable){
-      if(logStackTraceFromMessage){
-        def tI = new ThrowableInformation(loggingEvent.message as Throwable)
-        throwableAsString = tI.throwableStrRep
-      } else {
-        fullMessage.append(layout ? layout.format(loggingEvent) : (loggingEvent.getMessage() as String))
-      }
-    } else {
-      fullMessage.append(layout ? layout.format(loggingEvent) : loggingEvent.getMessage())
-    }
 
-    if (layout == null || layout.ignoresThrowable() || !fullMessage.length()) {
-      throwableAsString = throwableAsString?:loggingEvent.getThrowableStrRep()
-      if (throwableAsString != null) {
-        int len = throwableAsString.length
-        if(fullMessage.length()){ // newline after the 'original log message' to have nicer formatting
-          fullMessage.append Layout.LINE_SEP
-        }
-        for (int i = 0; i < len && i <  this.getMaxLoggedLines(); i++) {
-          fullMessage.append throwableAsString[i]
-          fullMessage.append Layout.LINE_SEP
-        }
-      }
-    }
```

Comment:
All of this and the code removed below has been moved into the `GelfMessageBuilder` to make it a bit more structured and remove duplicated code